swarm/src/lib.rs

```diff
@@ -1850,18 +1859,17 @@ mod tests {
                             },
                         );
                         state = State::Disconnecting;
+                        continue;
```

Comment:
Just above we provided a new event to be emitted by the behaviour. `continue` to ensure the behaviour is polled again.